libs/core/langchain_core/_api/deprecation.py

```diff
@@ -133,7 +133,13 @@ def deprecate(
         _package: str = package,
     ) -> T:
         """Implementation of the decorator returned by `deprecated`."""
-        from pydantic.v1.fields import FieldInfo  # pydantic: ignore
+        try:
+            # pydantic >= 1.10.17
+            from pydantic.v1.fields import FieldInfo  # pydantic: ignore
```

Comment:
Could we move to compat layer in langchain_core.utils.pydantic or make sure that we can accept FieldInfo that comes from either namespace and either version of pydantic? Otherwise this is just another thing that could break during 0.3